pkg/detectors/alibaba/alibaba.go

```diff
@@ -17,68 +24,109 @@ type Scanner struct{}
 var _ detectors.Detector = (*Scanner)(nil)
 
 var (
-	keyPat    = regexp.MustCompile(`\b(LTAI[a-zA-Z0-9]{17,21})[\"' ;\s]*`)
-	secretPat = regexp.MustCompile(`\b([a-zA-Z0-9]{30})\b`)
+	client = common.SaneHttpClient()
+
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+	keyPat = regexp.MustCompile(`\b([a-zA-Z0-9]{30})\b`)
+	idPat  = regexp.MustCompile(`\b(LTAI[a-zA-Z0-9]{17,21})[\"';\s]*`)
 )
 
 // Keywords are used for efficiently pre-filtering chunks.
 // Use identifiers in the secret preferably, or the provider name.
 func (s Scanner) Keywords() []string {
-	return []string{"LTAI"}
+	return []string{"alibaba"}
```

Comment:
We should use `LTAI` as the keyword here because it is a unique prefix that is always part of the ID pattern. This substring will always be present but `alibaba` may not, so it is a better choice.